Common/cpp/NativeModules/NativeReanimatedModule.h

```diff
@@ -156,22 +155,35 @@ class NativeReanimatedModule : public NativeReanimatedModuleSpec {
     return layoutAnimationsManager_;
   }
 
+  inline jsi::Runtime &getUIRuntime() {
+    return uiWorkletRuntime_->getRuntime();
+  }
+
  private:
+  void requestAnimationFrame(jsi::Runtime &rt, const jsi::Value &callback);
+
 #ifdef RCT_NEW_ARCH_ENABLED
   bool isThereAnyLayoutProp(jsi::Runtime &rt, const jsi::Object &props);
 #endif // RCT_NEW_ARCH_ENABLED
 
-  std::unique_ptr<EventHandlerRegistry> eventHandlerRegistry;
-  const RequestRenderFunction requestRender;
-  std::vector<FrameCallback> frameCallbacks;
-  bool renderRequested = false;
-  const ObtainPropFunction obtainPropFunction_;
-  std::function<void(double)> onRenderCallback;
-  AnimatedSensorModule animatedSensorModule;
-  ConfigurePropsFunction configurePropsPlatformFunction;
+  const std::shared_ptr<MessageQueueThread> jsQueue_;
+  const std::shared_ptr<JSScheduler> jsScheduler_;
+  const std::shared_ptr<UIScheduler> uiScheduler_;
+  std::shared_ptr<WorkletRuntime> uiWorkletRuntime_;
+  std::string valueUnpackerCode_;
+
+  std::unique_ptr<EventHandlerRegistry> eventHandlerRegistry_;
+  const RequestRenderFunction requestRender_;
+  std::vector<std::shared_ptr<jsi::Value>> frameCallbacks_;
+  volatile bool renderRequested_{false};
```

Comment:
We don't need an atomic here because `renderRequested_` is accessed only from one thread (i.e. UI thread).

I decided to add `volatile` keyword here to tell the compiler that the value of the variable may change at any time so it shouldn't be optimized. I don't think it's a must here but let's be on the safe side.